File: Low Level Design (LLD)/DesignPatterns/3. Structural/State/Cpp/src/DraftState.cpp
#include <iostream>
#include "../include/DraftState.h"
#include "../include/ModerationState.h"
#include "../include/Document.h"

void DraftState::edit(Document* doc) {
    std::cout << "Editing document in draft.\n";
}

void DraftState::publish(Document* doc) {
    std::cout << "Submitting document for review.\n";
    doc->setState(new ModerationState());
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/State/Java/DraftState.java
public class DraftState implements State {
    @Override
    public void edit(Document doc) {
        System.out.println("Editing document in draft.");
    }

    @Override
    public void publish(Document doc) {
        System.out.println("Submitting document for review.");
        doc.setState(new ModerationState());
    }
}
